TrackingTools/TrackAssociator/src/TrackDetectorAssociator.cc

```diff
@@ -979,10 +945,10 @@ void TrackDetectorAssociator::fillCaloTruth(const edm::Event& iEvent,
     info.ecalTrueEnergy = ecalTrueEnergy;
     info.hcalTrueEnergy = hcalTrueEnergy;
     info.hcalTrueEnergyCorrected = hcalTrueEnergy;
-    if (fabs(info.trkGlobPosAtHcal.eta()) < 1.3)
-      info.hcalTrueEnergyCorrected = hcalTrueEnergy * 113.2;
-    else if (fabs(info.trkGlobPosAtHcal.eta()) < 3.0)
-      info.hcalTrueEnergyCorrected = hcalTrueEnergy * 167.2;
+    if (std::abs(info.trkGlobPosAtHcal.eta()) < 1.3f)
+      info.hcalTrueEnergyCorrected = hcalTrueEnergy * 113.2f;
```

Comment:
This class contains quite several hardcoded parameters, which are prone to get out of date and hard to maintain. These caliibration factors for the hcal energy are one example (@abdoulline @igv4321 FYI), but there are several more around.
It is not the purpose of this PR, but I think that at some point one should make them either configurable, or taken from some central database or code.

Reply:
Well, these hardcoded HCAL scaling factors can be replaced with config.ones, but it requires some (importing) gymnastics: 

(1) here in  the  TrackDetectorAssociator.cc  from hit = simHcalHits  one can get  
HcalDetId id;   like this 
https://cmssdt.cern.ch/lxr/source/Validation/HcalHits/src/SimHitsValidationHcal.cc#0172
 
(2) 
int ieta = id.ieta();   int subdet = id.subdet();  
where the latter is 1 (HcalGenericDetId::HcalGenBarrel)  or 2 (HcalGenericDetId::HcalGenEndcap) as enumerated in 
https://cmssdt.cern.ch/lxr/source/DataFormats/HcalDetId/interface/HcalGenericDetId.h
 
(3) there are "Sampling factors" separately for HB and HE  (for approx. estimate of SimHits energy scale -> Reco scale used in HCAL Digitization):
https://cmssdt.cern.ch/lxr/source/SimCalorimetry/HcalSimProducers/python/hcalSimParameters_cfi.py#0066
https://cmssdt.cern.ch/lxr/source/SimCalorimetry/HcalSimProducers/python/hcalSimParameters_cfi.py#0085

for |ieta|=1-16  and subdet==HcalGenericDetId::HcalGenBarrel and 
for |ieta|=16-29 and subdet==HcalGenericDetId::HcalGenEndcap
as specified in 
https://cmssdt.cern.ch/lxr/source/SimCalorimetry/HcalSimProducers/python/hcalSimParameters_cfi.py#0062
https://cmssdt.cern.ch/lxr/source/SimCalorimetry/HcalSimProducers/python/hcalSimParameters_cfi.py#0081   
repectively